src/Microsoft.DotNet.Build.Tasks/PackageFiles/init-tools.ps1

```diff
@@ -56,10 +58,15 @@ if ($packageVersionPropsUrl)
                 $proxyCredentialsRequired = ($we -ne $null -and ([Net.HttpWebResponse]$we.Response).StatusCode -eq [Net.HttpStatusCode]::ProxyAuthenticationRequired)
                 Start-Sleep -Seconds (5 * $retryCount)
                 $retryCount++
-            } 
+            }
+            Write-Host "Failed to download '$packageVersionPropsPath'. Trying again..."
         }
     } while ($success -eq $false);
 
     Write-Host "Downloaded package version props:"
     Get-Content $packageVersionPropsPath
 }
+else
+{
+    Write-Host "'$packageVersionPropsUrl' url does not exist."
```

Comment:
if this is false ` if ($packageVersionPropsUrl)` then is it safe to write it out? I assume it is false because it doesn't exist or possibly is whitespace.
I suggest just logging "Environment variable PACKAGEVERSIONPROPSURL is not defined."

Also as a matter of style whenever you have a pattern liek
```
if (something)
{
.... big block of actual work ...
}
else
{
... small block returning early...
}
```

it is often easier to read if you change to
```
if (!something)
{
     ... small block return early...
}
... big block ...
```

This reduces indenting and makes it easier to follow what is going on when you get to the bottom of the code. Basically if you can early return, it is usually better. Rarely is there a reason not to (long ago, it was important to return only at the end of the function to help the compiler)